tools/syscall_header_gen.py

```diff
@@ -201,18 +193,36 @@ def convert_to_assembler_format(s):
     return '_'.join(words).upper()
 
 
+def map_syscalls_neg(syscalls):
+    # This function will map a list of (condition, syscall_list) tuples
+    # From: [(cond1, ["SyscallOne", "SysCallTwo"]),
+    #        (cond2, ["SyscallThree", ...])]
+    # Into: [(cond1, [("SyscallOne", -1), ("SyscallTwo", -2)]),
+    #        (cond2, [(SyscallThree", -3), ...])]
+    r = range(1, sys.maxsize)
+
+    def helper(syscalls):
+        nonlocal r
+        variants = zip(syscalls[1], map(lambda x: x * -1, r))
+        r = range(r.start + len(syscalls[1]), sys.maxsize)
+        return (syscalls[0], list(variants))
+    return map(helper, syscalls)
+
+
 def generate_kernel_file(kernel_header, api, debug):
     template = Environment(loader=BaseLoader, trim_blocks=False,
                            lstrip_blocks=False).from_string(KERNEL_HEADER_TEMPLATE)
-    data = template.render({'assembler': api, 'enum': api + debug,
-                            'upper': convert_to_assembler_format})
+    data = template.render({'assembler': list(map_syscalls_neg(api)),
+                            'enum': list(map_syscalls_neg(api + debug)),
+                            'upper': convert_to_assembler_format,
+                            'syscall_min': sum(map(lambda l: len(l[1]), api)) * -1})
```

Comment:
```suggestion
  'syscall_min': -sum(map(lambda l: len(l[1]), syscalls)) 
```
or use `-sum([ len(lst) for (cond, lst) in api ])` to avoid lambdas completely`